### PR TITLE
Move proxy-related fields to proxy config object

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
This matches the Prometheus data structure: 
- https://github.com/prometheus/common/blob/7043ea0e691b6da9ecbd08e7ae41e9cf28898e98/config/http_config.go#L1196

This is then used in the root HTTP Client and OAuth2 configs in Prometheus:
- https://github.com/prometheus/common/blob/7043ea0e691b6da9ecbd08e7ae41e9cf28898e98/config/http_config.go#L226
- https://github.com/prometheus/common/blob/7043ea0e691b6da9ecbd08e7ae41e9cf28898e98/config/http_config.go#L289

### CODE REVIEW

File: doc/api.md [hunk 1]
```diff
@@ -104,7 +105,7 @@ Authorization specifies a subset of the Authorization struct, that is safe for u
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
 | type | Set the authentication type. Defaults to Bearer, Basic will cause an error | string | false |
-| credentials | The secret's key that contains the credentials of the request | *[v1.SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#secretkeyselector-v1-core) | false |
+| credentials | The secret's key that contains the credentials of the request | *corev1.SecretKeySelector | false |
```
Comment (bernot-dev) [inline doc/api.md:108]: Interesting that the link got dropped. Core is just an alias...I wonder why that got carried forward here.

File: pkg/operator/apis/monitoring/v1/http.go [hunk 2]
```diff
@@ -37,3 +38,41 @@ func (c *TLS) ToPrometheusConfig() (*config.TLSConfig, error) {
 		MaxVersion:         maxVersion,
 	}, nil
 }
+
+func (c *ProxyConfig) ToPrometheusConfig() (config.URL, error) {
+	proxyURL, err := url.Parse(c.ProxyURL)
+	if err != nil {
+		return config.URL{URL: nil}, fmt.Errorf("invalid proxy URL: %w", err)
```
Comment (bernot-dev) [inline pkg/operator/apis/monitoring/v1/http.go:45]: Any reason to set `URL: nil` here? Should just be an empty literal, right?

**Review by bernot-dev — APPROVED** [review]
Non-blocking comments. LGTM.

File: doc/api.md (hunk 1 above)
Comment (TheSpiritXIII) [reply doc/api.md:108]: Huh, probably a bug in the underlying doc generator, which seems like it has been deleted and no longer maintained.

**Pull request merged**